vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/HasSelection.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.textfield;
```

Comment:
What do you think about moving this to the new shared library, so that no package name change is needed later, if e.g. other fields should implement this?

Reply:
Do you mean to `com.vaadin.flow.component` (whereever that is) like HasClearButton?

Reply:
Probably some other interfaces in textfield package should be moved there too (at som point)🤔

Reply:
Exactly :) If no objection are raised